example/main.go

```diff
@@ -128,3 +183,71 @@ func (r *reconcileReplicaSet) Reconcile(request reconcile.Request) (reconcile.Re
 
 	return reconcile.Result{}, nil
 }
+
+// podAnnotator annotates Pods
+type podAnnotator struct {
+	client client.Client
+}
+
+// Implement admission.Handler so the controller can handle admission request.
+var _ admission.Handler = &podAnnotator{}
+
+// podAnnotator adds an annotation to every incoming pods.
+func (a *podAnnotator) Handle(_ context.Context, req admission.Request) admission.Response {
+	pod := &corev1.Pod{}
+
+	err := admission.DecodeRequest(req, pod, nil)
```

Comment:
again, we should be using the codec from an injected scheme, or from the user's scheme.  We shouldn't reccomend using nil here.